### ISSUE TITLE
Twitter-API Error Blocked 403

### ISSUE THREAD

**Issue by alloc7260** [JustAnotherArchivist/snscrape#875]
### Describe the bug

Getting Blocked 403 error when trying to scrape tweets using TwitterSearchScraper

### How to reproduce

`
import snscrape.modules.twitter as sntwitter
from tqdm.notebook import tqdm # make your loops show a progress meter

# Passing query to TwitterSearchScraper
get_tweets = sntwitter.TwitterSearchScraper('machine learning').get_items()

# Create a attribute container 
tweet_list = []

for i, tweet in enumerate(tqdm(get_tweets, total = 10)):  ## ERROR HERE
    if i>9: break
    tweet_list.append(tweet)

`

### Expected behaviour

Should collect tweet objects in tweet_list

### Operating system

Ubuntu (Google Colab)

### Python version: output of `python3 --version`

Python 3.10.11

### snscrape version: output of `snscrape --version`

snscrape 0. 6.2.2023032 0

### Scraper

TwitterSearchScraper

### How are you using snscrape?

Module (`import snscrape.modules.something` in Python code)

### Backtrace

Error Log :

`ERROR:snscrape.base:Error retrieving https://api.twitter.com/2/search/adaptive.json?include_profile_interstitial_type=1&include_blocking=1&include_blocked_by=1&include_followed_by=1&include_want_retweets=1&include_mute_edge=1&include_can_dm=1&include_can_media_tag=1&include_ext_has_nft_avatar=1&include_ext_is_blue_verified=1&include_ext_verified_type=1&skip_status=1&cards_platform=Web-12&include_cards=1&include_ext_alt_text=true&include_ext_limited_action_results=false&include_quote_count=true&include_reply_count=1&tweet_mode=extended&include_ext_collab_control=true&include_ext_views=true&include_entities=true&include_user_entities=true&include_ext_media_color=true&include_ext_media_availability=true&include_ext_sensitive_media_warning=true&include_ext_trusted_friends_metadata=true&send_error_codes=true&simple_quoted_tweet=true&q=adani+lang%3Aen+since%3A2023-01-26+until%3A2023-01-27&tweet_search_mode=live&count=20&query_source=spelling_expansion_revert_click&pc=1&spelling_corrections=1&include_ext_edit_control=true&ext=mediaStats%2ChighlightedLabel%2ChasNftAvatar%2CvoiceInfo%2Cenrichments%2CsuperFollowMetadata%2CunmentionInfo%2CeditControl%2Ccollab_control%2Cvibe: blocked (403)
CRITICAL:snscrape.base:4 requests to https://api.twitter.com/2/search/adaptive.json?include_profile_interstitial_type=1&include_blocking=1&include_blocked_by=1&include_followed_by=1&include_want_retweets=1&include_mute_edge=1&include_can_dm=1&include_can_media_tag=1&include_ext_has_nft_avatar=1&include_ext_is_blue_verified=1&include_ext_verified_type=1&skip_status=1&cards_platform=Web-12&include_cards=1&include_ext_alt_text=true&include_ext_limited_action_results=false&include_quote_count=true&include_reply_count=1&tweet_mode=extended&include_ext_collab_control=true&include_ext_views=true&include_entities=true&include_user_entities=true&include_ext_media_color=true&include_ext_media_availability=true&include_ext_sensitive_media_warning=true&include_ext_trusted_friends_metadata=true&send_error_codes=true&simple_quoted_tweet=true&q=adani+lang%3Aen+since%3A2023-01-26+until%3A2023-01-27&tweet_search_mode=live&count=20&query_source=spelling_expansion_revert_click&pc=1&spelling_corrections=1&include_ext_edit_control=true&ext=mediaStats%2ChighlightedLabel%2ChasNftAvatar%2CvoiceInfo%2Cenrichments%2CsuperFollowMetadata%2CunmentionInfo%2CeditControl%2Ccollab_control%2Cvibe failed, giving up.
CRITICAL:snscrape.base:Errors: blocked (403), blocked (403), blocked (403), blocked (403)
---------------------------------------------------------------------------
ScraperException                          Traceback (most recent call last)
<ipython-input-3-623ad7c14973> in <cell line: 9>()
      7 tweet_list = []
      8 
----> 9 for i, tweet in enumerate(tqdm(get_tweets, total = 10)):
     10     if i>9: break
     11     tweet_list.append(tweet)

6 frames
/usr/local/lib/python3.10/dist-packages/tqdm/notebook.py in __iter__(self)
    252         try:
    253             it = super(tqdm_notebook, self).__iter__()
--> 254             for obj in it:
    255                 # return super(tqdm...) will not catch exception
    256                 yield obj

/usr/local/lib/python3.10/dist-packages/tqdm/std.py in __iter__(self)
   1176 
   1177         try:
-> 1178             for obj in iterable:
   1179                 yield obj
   1180                 # Update and possibly print the progressbar.

/usr/local/lib/python3.10/dist-packages/snscrape/modules/twitter.py in get_items(self)
   1659                 del params['cursor']
   1660 
-> 1661                 for obj in self._iter_api_data('https://api.twitter.com/2/search/adaptive.json', _TwitterAPIType.V2, params, paginationParams, cursor = self._cursor):
   1662                         yield from self._v2_timeline_instructions_to_tweets_or_users(obj)
   1663 

/usr/local/lib/python3.10/dist-packages/snscrape/modules/twitter.py in _iter_api_data(self, endpoint, apiType, params, paginationParams, cursor, direction)
    759                 while True:
    760                         _logger.info(f'Retrieving scroll page {cursor}')
--> 761                         obj = self._get_api_data(endpoint, apiType, reqParams)
    762                         yield obj
    763 

/usr/local/lib/python3.10/dist-packages/snscrape/modules/twitter.py in _get_api_data(self, endpoint, apiType, params)
    725                 if apiType is _TwitterAPIType.GRAPHQL:
    726                         params = urllib.parse.urlencode({k: json.dumps(v, separators = (',', ':')) for k, v in params.items()}, quote_via = urllib.parse.quote)
--> 727                 r = self._get(endpoint, params = params, headers = self._apiHeaders, responseOkCallback = self._check_api_response)
    728                 try:
    729                         obj = r.json()

/usr/local/lib/python3.10/dist-packages/snscrape/base.py in _get(self, *args, **kwargs)
    249 
    250         def _get(self, *args, **kwargs):
--> 251                 return self._request('GET', *args, **kwargs)
    252 
    253         def _post(self, *args, **kwargs):

/usr/local/lib/python3.10/dist-packages/snscrape/base.py in _request(self, method, url, params, data, headers, timeout, responseOkCallback, allowRedirects, proxies)
    245                         _logger.fatal(msg)
    246                         _logger.fatal(f'Errors: {", ".join(errors)}')
--> 247                         raise ScraperException(msg)
    248                 raise RuntimeError('Reached unreachable code')
    249 

ScraperException: 4 requests to https://api.twitter.com/2/search/adaptive.json?include_profile_interstitial_type=1&include_blocking=1&include_blocked_by=1&include_followed_by=1&include_want_retweets=1&include_mute_edge=1&include_can_dm=1&include_can_media_tag=1&include_ext_has_nft_avatar=1&include_ext_is_blue_verified=1&include_ext_verified_type=1&skip_status=1&cards_platform=Web-12&include_cards=1&include_ext_alt_text=true&include_ext_limited_action_results=false&include_quote_count=true&include_reply_count=1&tweet_mode=extended&include_ext_collab_control=true&include_ext_views=true&include_entities=true&include_user_entities=true&include_ext_media_color=true&include_ext_media_availability=true&include_ext_sensitive_media_warning=true&include_ext_trusted_friends_metadata=true&send_error_codes=true&simple_quoted_tweet=true&q=adani+lang%3Aen+since%3A2023-01-26+until%3A2023-01-27&tweet_search_mode=live&count=20&query_source=spelling_expansion_revert_click&pc=1&spelling_corrections=1&include_ext_edit_control=true&ext=mediaStats%2ChighlightedLabel%2ChasNftAvatar%2CvoiceInfo%2Cenrichments%2CsuperFollowMetadata%2CunmentionInfo%2CeditControl%2Ccollab_control%2Cvibe failed, giving up.`

**Comment by devshapanara** [comment]
yeah I'm also dizzy over this, do you guys have a solution for this

**Comment by juanluidos** [comment]
#846 Take a look at this issue
